app/scripts/modules/core/src/application/nav/NavRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { UIRouterContextComponent } from '@uirouter/react-hybrid';
+import { UISref } from '@uirouter/react';
+
+import { NavCategory } from './NavCategory';
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../../application';
+
+export interface INavRouteProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavRoute = ({ app, category, isActive }: INavRouteProps) => (
+  <UIRouterContextComponent key={category.key}>
```

Comment:
`key` is not necessary here, right?

Reply:
Yes I was clearing a React warning since multiple `NavRoutes` are rendered.

Reply:
the `key` should be added to the `NavRoute` when mapping over them `array.map(category => <NavRoute key={category.key} />)`